src/cdi_CPEloss/Tabular_CP_Eloss.hh

```diff
@@ -0,0 +1,113 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!                                                                              
+ * \file   cdi_CPEloss/Tabular_CP_Eloss.hh                                     
+ * \author Ben R. Ryan                                                        
+ * \date   2019 Nov 4                                              
+ * \brief  Tabular_CP_Eloss class definition.                                   
+ * \note   Copyright (C) 2016-2019 Triad National Security, LLC.                 
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#ifndef __cdi_CPEloss_Tabular_CP_Eloss_hh__
+#define __cdi_CPEloss_Tabular_CP_Eloss_hh__
+
+#include "experimental/mdspan"
+#include "cdi/CPCommon.hh"
+#include "cdi/CPEloss.hh"
+#include "ds++/Assert.hh"
+#include "ds++/DracoMath.hh"
+#include "units/PhysicalConstants.hh"
+#include "units/UnitSystemType.hh"
+#include <cmath>
+#include <fstream>
+#include <string>
+
+namespace rtt_cdi_cpeloss {
+//===========================================================================//
+/*!                                                                              
+ * \class Tabular_CP_Eloss                                                      
+ *                                                                               
+ * \brief Derived rtt_cdi::CPEloss class for tabular eloss.                     
+ *        This class implements the interface found in cdi/CPEloss.hh for        
+ *        the case where CP energy loss data is in tabular form, stored 
+ *        in a file.                
+ *                                                                               
+ */
+//===========================================================================//
+
+class Tabular_CP_Eloss : public rtt_cdi::CPEloss {
+public:
+  typedef std::vector<double> sf_double;
```

Comment:
Can you add a couple of typedefs here maybe:
`typedef std::experimental::dynamic_extent dyanmic_extent;`
`typedef std::experimental::layout_left layout_left;`
This should make line 70 a little more clear.

Reply:
Another option (if you want to keep the namespace scope) would be `typedef std::experimental stdex;`. I think this is what the kokkos folks did in the test.

Reply:
@KineticTheory `typedef std::experimental stdex;` didn't work for me, either at the namespace level or inside the class definition. What did work for me was `namespace stdex = std::experimental;` inside the `rtt_cdi_cpeloss` namespace in `Tabular_CP_Eloss.hh`; highlighting this here in case that is not OK.